From d46834808c3226b3a6e48649df65befc399c21cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Lyson=C4=9Bk?= <olysonek@redhat.com>
Date: Wed, 11 Jul 2018 00:41:45 +0200
Subject: [PATCH] tuned-gui: Sort plugins based on their name
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Previously the sorting was done by comparing the objects themselves,
which is not what we want and it doesn't work in Python 3 - TypeError
is raised, e.g:
TypeError: '<' not supported between instances of 'BootloaderPlugin' and 'MountsPlugin'

Signed-off-by: Ondřej Lysoněk <olysonek@redhat.com>
---
 tuned-gui.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tuned-gui.py b/tuned-gui.py
index 8f72fd5..e486687 100755
--- a/tuned-gui.py
+++ b/tuned-gui.py
@@ -278,7 +278,8 @@ class Base(object):
 		self.treestore_profiles = Gtk.ListStore(GObject.TYPE_STRING,
 				GObject.TYPE_STRING)
 		self.treestore_plugins = Gtk.ListStore(GObject.TYPE_STRING)
-		for plugin in sorted(self.plugin_loader.plugins):
+		for plugin in sorted(self.plugin_loader.plugins,
+				key = lambda plugin: plugin.name):
 			self.treestore_plugins.append([plugin.name])
 		self.combobox_plugins = \
 			self.builder.get_object('comboboxPlugins')
-- 
2.14.4

